RFC/RFC-14-p2p-adapter.md

```diff
@@ -17,24 +17,39 @@ This document describes a possible implementation for a P2P adapter using a fall
 
 A P2P adapter is needed in order to provide a light mechanism for comms (compared to [RFC-5: comms service using websocket](/RFC/RFC-5-ws-comms.md)). This adapter should guarantee the delivery of a message to any peer in the network (or cluster, [real-time clustering of users in Islands (ADR-35)](/adr/ADR-35)).
 
-In this approach a Routing Table is used, which is managed by the Routing Service. The idea is to limit the amount of times that any peer receives a packet to 1, currently with naive broadcasting p2p solution each node relays a package only once but it can receive as many times as peer conections has.
+The new implementation should solve the problems in the old version[link]:
+
+- When two peers are unable to see each other, it's hard to identify if there is a network failure or a cluster or a bug in the client.
+- Since all peers broadcast information on an ever changing logical mesh (every tick the relay changes), it's almost impossible to measure latency. We also don't know the impact of the amount of packages that are circulating around the network or even how many are been currently relayed. The relay suspension mechanism that prevents the network to be flooded with packages also prevents the stabilization of the mesh.
+- We don't have a way to measure the impact of the "suspension relay" heuristic, which means we don't know if the available routes are efficent.
+- There are two unhandled scenarios which may cause messages to not be deliver:
+  - if there are two clusters (or more), peers will be completely isolated between each other, no communication using the P2P mesh is possible.
+  - At some point, relay suspension may suspend the only peer that is able to deliver a message. This is a temporal failure, since the next time the mesh is negotiated, the delivery will be resumed. 
 
 ## Approach
 
-This approach is simple, it should be easy to debug and it could be refined in the future for performance if needed. 
+This approach focuses on addressing the problems mentioned above:
+  1. Handle clusters: identifing peers outside of the current mesh, and providing a fallback service to deliver the messages. Optionally (outside of the scope of this RFC), this service could provide hints to the peers in order to connect separated clusters.
+  2. Efficient usage of resources: messages are received only once, they are decoded only once, they are relayed using only the most efficent route and thus avoiding flooding the network. There is also no additional expiration ttl, or hops ttl since the route is already determined.
```

Comment:
```suggestion
  2. Efficient usage of resources: messages are received only once, they are decoded only once, and they are relayed using only the most efficient route and thus avoiding flooding the network. There is also no additional expiration TTL or hops TTL since the route is already determined.
```